rust_src/src/eval.rs

```diff
@@ -860,4 +861,115 @@ pub fn autoload_do_load(
     }
 }
 
+/// Run each hook in HOOKS.
+/// Each argument should be a symbol, a hook variable.
+/// These symbols are processed in the order specified.
+/// If a hook symbol has a non-nil value, that value may be a function
+/// or a list of functions to be called to run the hook.
+/// If the value is a function, it is called with no arguments.
+/// If it is a list, the elements are called, in order, with no arguments.
+///
+/// Major modes should not use this function directly to run their mode
+/// hook; they should use `run-mode-hooks' instead.
+///
+/// Do not use `make-local-variable' to make a hook variable buffer-local.
+/// Instead, use `add-hook' and specify t for the LOCAL argument.
+/// usage: (run-hooks &rest HOOKS)
+#[lisp_fn]
+pub fn run_hooks(args: &mut [LispObject]) -> () {
+    for item in args {
+        run_hook(item.to_raw());
+    }
+}
+
+/// Run HOOK with the specified arguments ARGS.
+/// HOOK should be a symbol, a hook variable.  The value of HOOK
+/// may be nil, a function, or a list of functions.  Call each
+/// function in order with arguments ARGS.  The final return value
+/// is unspecified.
+///
+/// Do not use `make-local-variable' to make a hook variable buffer-local.
+/// Instead, use `add-hook' and specify t for the LOCAL argument.
+/// usage: (run-hook-with-args HOOK &rest ARGS)
+#[lisp_fn]
+pub fn run_hook_with_args(args: &mut [LispObject]) -> LispObject {
+    run_hook_with_args_internal(args, funcall_nil)
+}
+
+fn funcall_nil(args: &mut [LispObject]) -> LispObject {
+    let mut obj_array: Vec<Lisp_Object> = args.iter().map(|o| o.to_raw()).collect();
+    unsafe { Ffuncall(obj_array.len() as isize, obj_array.as_mut_ptr()) };
+    LispObject::constant_nil()
+}
+
+/* Run the hook HOOK, giving each function no args.  */
```

Comment:
For consistency, we should probably use `///` comments here.

Reply:
right. Oops.